yarn-project/canary/src/utils.ts

```diff
@@ -129,3 +147,317 @@ export async function deployL1Contract(
 
   return EthAddress.fromString(receipt.contractAddress!);
 }
+
+/**
+ * A Class for testing cross chain interactions, contains common interactions
+ * shared between cross chain tests.
+ */
+export class CrossChainTestHarness {
+  static async new(
```

Comment:
had to modify the `CrossChainTestHarness.new()` a bit to make it workable on sandbox. Otherwise it is mostly a copy paste!